persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -120,7 +121,9 @@ public static HashMap<String, String> readToml(Path configPath, String name) thr
             for (Object member : nodeList) {
                 if (member instanceof TableNode) {
                     TableNode node = (TableNode) member;
-                    if (node.identifier().toSourceCode().trim().equals(name)) {
+                    if (node.identifier().toSourceCode().trim().equals(name) ||
+                            ((node.identifier().toSourceCode().trim().equals(DATABASE)) &&
+                                    (configPath.toString().contains(PERSIST_TOML_FILE)))) {
```

Comment:
What is the logic here? What is the name you are checking most importantly why the table title contains Persit toml file name?

Reply:
The code should be readable that means anyone should understand the logic knowing the feature description with little context.

Reply:
What happens here is, we check the table name to see if the config file has [project_name.client] table or [database] table. but the second case is only happen's if the file name is persist. Since we cannot assure the config file won't have [database] table we have to check whether the file is `persist.toml`

Reply:
After the design change, Is it necessary for us to check if Config.toml contains all 5 correct configs? It will only come into place if we template all entries in Persist.toml
```
Persist.toml
[database]
user="asdad"
password="qwwd"
host="localhost"
port=3306
database="$foo.clients.database"

Config.toml
[foo.client.database]
database="customer"
```
How will this logic behave?  

When the functionality changes, sometimes we cant keep the same functions. We have to change it as needful.

Reply:
This will return an error as follows
```Persist client related config doesn't exist in Config.toml.
You should add database configurations under <org name>.<pkg name>.clients 
```

Reply:
> After the design change, Is it necessary for us to check if Config.toml contains all 5 correct configs? It will only come into place if we template all entries in Persist.toml
> 
> ```
> Persist.toml
> [database]
> user="asdad"
> password="qwwd"
> host="localhost"
> port=3306
> database="$foo.clients.database"
> 
> Config.toml
> [foo.client.database]
> database="customer"
> ```
> 
> How will this logic behave?

This will return an error as follows as the table name in config.toml doesn't comply with the standard
```
Persist client related config doesn't exist in Config.toml.
You should add database configurations under <org name>.<pkg name>.clients 
```

Reply:
But essentially from db_push command point of view. I don't need to have all configs in Config.toml. Do I? I am not running the project, I want to just setup a database. These configs are only needed to run the project

Reply:
Why do we need to do this condition `configPath.toString().contains(PERSIST_TOML_FILE)` check here? This should verify before calling this function. This function's task is to check the key for the provided config file and return if it exists. If not, return an error.

Reply:
Possibilities that can configure in config.toml file is,

1. with `<moduleName>.clients`
```
[foo.client]
database="customer"
```

2. with `<orgName>.<moduleName>.clients`
```
[foo.bar.client.database]
database="customer"
```

3. Have different configs from the runtime configs. and point the config in Persist.toml file.
```
toolDatabase="customer"
```

All these options should support by our tool.

Reply:
@sahanHe @niveathika any other possibilities?

Reply:
We only allow one templating isnt it? For instance can I point to a different table in Config.toml?

Reply:
Ideally, we are restricting to Config.toml only. Users can have different tables in the Config.toml. We can't give an error if the user points to a different valid table. How hard is it to give the value for the key from toml file.

Reply:
Should we support getting values from two tables. For instance we get 'port' from foo.clients and we get host from foo1.foo.clients

Reply:
updated the implementation to support this in [efc8d8e](https://github.com/ballerina-platform/persist-tools/pull/54/commits/efc8d8e0fa21126e5d1b92c00eded10671e1324e) and changed a test case in [ba26895](https://github.com/ballerina-platform/persist-tools/pull/54/commits/ba26895bc4e3fa77ccc140695effc73fd9067211)